pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -2424,7 +2433,10 @@ func buildHostnameMatch(localNamespace string, r GatewayResources, l k8s.Listene
 
 	resp := []string{}
 	for _, ns := range namespacesFromSelector(localNamespace, r, l.AllowedRoutes) {
-		resp = append(resp, fmt.Sprintf("%s/%s", ns, hostname))
+		// This check is necessary to prevent adding a hostname with an invalid empty namespace
+		if len(ns) > 0 {
+			resp = append(resp, fmt.Sprintf("%s/%s", ns, hostname))
+		}
```

Comment:
Excluding the `""` empty string namespace picked up by tracking `"*"` when building the Istio Gateway server hosts array was the trick needed to fix the invalid generated CRDs.